### PR TITLE
Muse Dash: Add filler items and rework generation balance

### PR DESCRIPTION
**Pull request by DeamonHunter**
## What is this fixing or adding?
This PR aims to fix a problem within Muse Dash. The total lack of filler items. This ends up changing the balance of the Muse Dash rando so other changes have been made.

Full list of changes:
- Added 3 filler items. These are 1 time consumables to Muse Dash.
- Made all songs have 2 locations. (Rather than a percentage of them)
  - This was to simplify balance. (Having a variable amount of songs meant that Muse Dash could be generated without filler.)
- When filling the multiworld the following changes were made:
  - Music Sheet count has been slightly lowered. (No Longer double counts starting songs.)
  - After adding music sheets and the initial lot of songs (and traps), 50% of the remaining spots are now filled with filler.
  - The final remaining slots are filled with duplicate songs. These songs are no longer progressive.

## How was this tested?
Using multiple Beta Test games using discord. As well as existing tests to ensure no old functionality was broken.

### CODE REVIEW

File: worlds/musedash/Options.py [hunk 1]
```diff
@@ -101,20 +103,10 @@ class GradeNeeded(Choice):
...
-    display_name = "Additional Item %"
-    range_start = 50
-    default = 80
-    range_end = 100
-
-
 class MusicSheetCountPercentage(Range):
-    """Collecting enough Music Sheets will unlock the goal song needed for completion.
-    This option controls how many are in the item pool, based on the total number of songs."""
+    """Controls how many music sheets to the pool based on the number of songs, including starting songs.
```
Comment (nicholassaylor) [inline worlds/musedash/Options.py:107]:
```suggestion
    """Controls how many music sheets are added to the pool based on the number of songs, including starting songs.
```

**Review by nicholassaylor — CHANGES_REQUESTED** [review]
Options and presets look good except for a small typo in the tooltip marked in the review.

Only reviewed Options and Presets

File: worlds/musedash/Options.py (hunk 1 above)
Comment (DeamonHunter) [reply worlds/musedash/Options.py]: Thanks!

**Review by benny-dreamly — APPROVED** [review]

**Review by qwint — APPROVED** [review]
Reviewed code updates and everything I was concerned about turned out to be already covered, (like create_items() seemingly being ok with adding more items than locations, but not being mathematically possible)

Also ran generate on different song count, trap settings, and music sheet percents (making sure to confirm both none trap types and 0 trap percent don't add any traps to the pool) to confirm no fill errors are weird values,
but did not play any games as I don't own it.

One thing that shouldn't stop this PR at all but is good to note is the TestDifficultyRanges test file is still using multiworld.option, even though the rest of the world is up to date.

That and i'm assuming the Unit test failure is a random failure, nothing failed on my machine (but I'm not running the same profile that failed)

**Comment by nicholassaylor** [comment]
> That and i'm assuming the Unit test failure is a random failure, nothing failed on my machine (but I'm not running the same profile that failed)

Failure is from MMBN3 and is unrelated to the PR (for those concerned with the failure)

**Comment by DeamonHunter** [comment]
Thanks for catching the old options usage. And yes the old test failure was unrelated to the PR.

It should be impossible for the create items func to create more items than locations. It does seem to not care at first, but that is because all worlds will have enough space for a copy of every song, plus all the music sheets required. (As every song has 2 locations now.) And then every step after that checks to ensure it doesn't add too much.

**Comment by DeamonHunter** [comment]
Resolved the merge conflict. Bringing this to core-review.

Special note if merging before the big async. This does end up removing the option `additional_item_percentage` which doesn't affect too much, but might need a bit of manual adjusting of yamls.

**Review by nicholassaylor — APPROVED** [review]
Changes I requested were fixed

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**